comp/logs/agent/config/config_keys.go

```diff
@@ -109,16 +109,28 @@ func (l *LogsConfigKeys) devModeUseProto() bool {
 	return l.getConfig().GetBool(l.getConfigKey("dev_mode_use_proto"))
 }
 
+// compressionKind returns the compression kind for the logs agent
+// However, other pipelines (agent_telemetry, databse_monitoring,network_device, container,sbom, service_discovery, compliance, runtime_security) also use this function.
+// Todo: Refactor this to eventually account for different config/global keys for different pipelines other than the logs pipeline.
 func (l *LogsConfigKeys) compressionKind() string {
-	compressionKind := l.getConfig().GetString(l.getConfigKey("compression_kind"))
-	switch compressionKind {
-	case "zstd", "gzip":
+	configKey := l.getConfigKey("compression_kind")
+	compressionKind := l.getConfig().GetString(configKey)
+
+	if l.prefix != "logs_config." {
```

Comment:
(I'm working on runtime security, so I'm impacted by this PR). I don't understand it at all. If I specifically ask for `zstd` in my prefix specific config, it's not going to be actually used ?
Why not using some `IsConfigured` or similar to inherit when it's not defined in the config ?
Also why should this be done just for the compression and not for other configs: like NoSSL, or the compression level ?